snippets/scripts/jira-components-to-compass-components/migrateJiraComponentsToCompassComponents.py

```diff
@@ -0,0 +1,302 @@
+      componentDetails {
+        id
+      }
+    }
+  }
+}
+"""
+
+TENANT_INFO_QUERY = """
+query tenantInfo($hostNames: [String!]) {
+  tenantContexts(hostNames:$hostNames) {
+    cloudId
+  }
+}
+"""
+
+
+def make_api_call(url, method, data=None, headers=None, no_response=False):
+    # print(f"Making a {method} request to {url}")
+    try:
+        response = requests.request(
+            method,
+            url,
+            auth=HTTPBasicAuth(USER_NAME, API_TOKEN),
+            json=data,
+            headers=headers
+        )
+        response.raise_for_status()
+        if no_response:
+            return
+        return response.json()
+    except HTTPError as http_err:
+        if http_err.response.status_code == 404:
+            return None
+        print(f"HTTP error occurred: {http_err} - {http_err.response.text}")
+        # rethrow the exception
+        raise http_err
+    except Exception as err:
+        print(f"An error occurred: {err}")
+
+
+def make_gql_api_call(query, variables):
+    url = f"https://{DOMAIN_NAME}/gateway/api/graphql"
+    headers = {
+        'Content-Type': 'application/json'
+    }
+    data = {
+        "query": query,
+        "variables": variables
+    }
+    return make_api_call(url, "POST", data, headers)
+
+
+def get_jira_component_list(project_key):
+    url = f"https://{DOMAIN_NAME}/rest/api/3/project/{project_key}/components"
+    response = make_api_call(url, "GET")
+    return response
+
+
+def get_related_issues_for_component(project_key, component):
+    component_id = component['id']
+    component_name = component['name']
+    params = {'jql': f"project = {project_key} AND component = '{component_name}'"}
+
+    start_at = 0
+    url = f"https://{DOMAIN_NAME}/rest/api/3/search?{urllib.parse.urlencode(params)}&startAt={start_at}"
+    all_issues = []
+    while True:
+        response = make_api_call(url, "GET")
+        issues = response['issues']
+        total = response['total']
+        start_at += len(issues)
+        all_issues.extend(issues)
+        if start_at >= total:
+            break
+        url = f"https://{DOMAIN_NAME}/rest/api/3/search?{urllib.parse.urlencode(params)}&startAt={start_at}"
+
+    # the jql api will return issues by component name
+    # which can include compass components with same name as well
+    # we need to filter out the ones that has the current Jira component in them
+    all_issues = [issue for issue in all_issues if any(component_id in component['id'] for component in issue['fields']['components'])]
+
+    return all_issues
+
+
+def get_tenant_info():
+    response = make_gql_api_call(TENANT_INFO_QUERY, {"hostNames": [DOMAIN_NAME]})
+    return response['data']['tenantContexts'][0]['cloudId']
+
+
+def create_compass_component(cloud_id,component):
+    url = f"https://{DOMAIN_NAME}/gateway/api/graphql"
+    headers = {
+        'Content-Type': 'application/json'
+    }
+    variables = {
+        "cloudId": cloud_id,
+        "input": {
```

Comment:
would it be a good idea to add a label to distinguish these components, like we do with other import methods?

Reply:
also, as it is now, it would create duplicate components if the script is re-run, correct? since I don't see a search query present

Reply:
> it would create duplicate components

We are actually not searching in Compass, we are searching in Jira! Once component is created in Compass, it should be synced in Jira and be available in the `/component` api. That way we get the `id` of the component in Jira as well to be able to make the update on the issue

Reply:
aah I missed that, makes sense 👍